common/tools.py

```
@@ -2006,49 +2008,53 @@ def __init__(self, callback = None, overwrite = True):
         self.overwrite = overwrite
 
     def start(self, timeout):
-        """
-        Start the timer (which calls the handler function
+        """Start the timer (which calls the handler function
         when the timer ran down).
 
-        The start is silently ignored if the current timer is still
-        ticking and the the attribute ``overwrite`` is ``False``.
+        If `self.overwrite` is ``False`` and the current timer is still ticking
+        the start is silently ignored.
 
         Args:
-            timeout: timer count down in seconds
+            timeout: Timer count down in seconds.
         """
         if self.ticking and not self.overwrite:
             return
+
         try:
             # Warning: This code may cause non-deterministic RTEs
             #          if the handler function calls code that does
             #          not support reentrance (see e.g. issue #1003).
             signal.signal(signal.SIGALRM, self.handler)
             signal.alarm(timeout)
         except ValueError:
+            # Why???
```

ValueErrors can be raised from [signal.signal](https://docs.python.org/3/library/signal.html#signal.signal). Doesn't explain _why_ we suppress the errors.

---

Maybe remove that in the future. I just didn't want to modify to much in one step. Who know what problem this can cause!?

---

Yes, I'd leave it for now.